vision/google/cloud/vision/image.py

```diff
@@ -30,22 +30,28 @@ class Image(object):
     :type content: bytes
     :param content: Byte stream of an image.
 
+    :type filename: str
+    :param filename: Filename to image.
+
     :type source_uri: str
     :param source_uri: Google Cloud Storage URI of image.
 
     :type client: :class:`~google.cloud.vision.client.Client`
     :param client: Instance of Vision client.
     """
 
-    def __init__(self, client, content=None, source_uri=None):
+    def __init__(self, client, content=None, filename=None, source_uri=None):
         self.client = client
-        self._content = None
-        self._source = None
 
-        if source_uri:
-            self._source = source_uri
-        else:
-            self._content = _bytes_to_unicode(b64encode(_to_bytes(content)))
+        if filename:
```

Comment:
Always remember your None-checks: `if filename is not None`

![nunchucks](https://cloud.githubusercontent.com/assets/520669/20739994/80991f68-b674-11e6-9aa7-bcc176705f0b.jpg)